### PR TITLE
Feature/em 2874 make bookmarks togglable

### PR DESCRIPTION
**Pull request by benouaer**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EM-2874

### Change description ###
make bookmarks togglable along with annotations, and fix rotation issues

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/store/actions/bookmarks.action.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
 import { Action } from '@ngrx/store';
-import { Bookmark } from '../reducers/bookmarks.reducer';
+import { Bookmark, PdfPosition } from '../../viewers/pdf-viewer/side-bar/bookmarks/bookmarks.interfaces';
 
 export const LOAD_BOOKMARKS = '[Bookmarks] Load Bookmarks';
 export const LOAD_BOOKMARKS_SUCCESS = '[Bookmarks] Load Bookmarks Success';
-export const LOAD_BOOKMARKS_FAIL = '[Bookmarks] Load Bookmarks Failure';
+export const LOAD_BOOKMARKS_FAILURE = '[Bookmarks] Load Bookmarks Failure';
```
Comment (fazanki) [inline projects/media-viewer/src/lib/store/actions/bookmarks.action.ts:6]: i think we are using fail everywhere else

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
               private bookmarksApiService: BookmarksApiService) {}
 
   @Effect()
   loadBookmarks$ = this.actions$.pipe(
     ofType(bookmarksActions.LOAD_BOOKMARKS),
-    map((action: bookmarksActions.LoadBookmarks) => action.payload),
-    exhaustMap((url) =>
-      this.bookmarksApiService.getBookmarks(url)
+    withLatestFrom(this.store.pipe(select(fromAnnotations.getDocumentIdSetId))),
+    map(([,docSetId]) => docSetId.documentId),
```
Comment (fazanki) [inline projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:25]: is not recommended to be connecting store to effect but rather u should pass payload (document id )in the actions plus you are not unsubscribing which is a memory leak

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
+    exhaustMap((documentId) =>
+      this.bookmarksApiService.getBookmarks(documentId)
```
Comment (fazanki) [inline projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:27]: this should go in separate effect - dispatch action and just call that effect

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts [hunk 4]
```diff
@@ -27,6 +35,21 @@ export class BookmarksEffects {
   createBookmark$ = this.actions$.pipe(
     ofType(bookmarksActions.CREATE_BOOKMARK),
     map((action: bookmarksActions.CreateBookmark) => action.payload),
+    withLatestFrom(
+      this.store.pipe(select(fromAnnotations.getDocumentIdSetId)),
+      this.store.pipe(select(fromBookmarks.getPdfPosition))
```
Comment (fazanki) [inline projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:40]: u can't connect here store its wrong implementation of ngrx... pass the payoad through action

File: projects/media-viewer/src/lib/store/store-utils.ts [hunk 5]
```diff
@@ -89,13 +90,10 @@ export class StoreUtils {
     }, {});
   }
 
-  static snakeCase = string => {
```
Comment (fazanki) [inline projects/media-viewer/src/lib/store/store-utils.ts:92]: is this not used any more

File: projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<div class="findbar doorHanger" [ngClass]="{ hidden: toolbarButtons.searchBarHidden | async }">
+<div class="findbar doorHanger" [ngClass]="{ hidden: toolbarEvents.searchBarHidden | async }">
```
Comment (fazanki) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/search-bar/search-bar.component.html:1]: u can use [hidden] default angular pipe

File: projects/media-viewer/src/lib/toolbar/main-toolbar/sub-toolbar/sub-toolbar.component.html [hunk 7]
```diff
@@ -1,5 +1,5 @@
 <div id="secondaryToolbar" class="secondaryToolbar doorHangerRight"
-     [ngClass]="{ hidden: toolbarButtons.subToolbarHidden | async }">
+     [ngClass]="{ hidden: toolbarEvents.subToolbarHidden | async }">
```
Comment (fazanki) [inline projects/media-viewer/src/lib/toolbar/main-toolbar/sub-toolbar/sub-toolbar.component.html:2]: use [hidden] remove css

File: projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/side-bar.component.ts [hunk 8]
```diff
@@ -138,29 +51,8 @@ export class SideBarComponent implements OnInit, OnChanges, OnDestroy {
...
-      yCoordinate: pdfLocation.top
-    } as Bookmark);
-  }
-
-  private extractDocumentId(url: string): string {
-    url = url.includes('/documents/') ? url.split('/documents/')[1] : url;
-    return url.replace('/binary', '');
+  onAddBookmarkClick() {
+    this.toggleSidebarView('bookmark');
+    this.store.dispatch(new CreateBookmark('new bookmark'));
```
Comment (fazanki) [inline projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/side-bar.component.ts:56]: whats 'new bookmark'

File: projects/media-viewer/src/lib/store/actions/bookmarks.action.ts (hunk 1 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/store/actions/bookmarks.action.ts:6]: I've been using failure in bookmarks, but I can change it back if we want it to be consistent everywhere

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts (hunk 3 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:27]: why, the only use for this effect is to get the bookmarks

File: projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts (hunk 4 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/store/effects/bookmarks.effects.ts:40]: I'm not sure why, I've used this pattern before, it's even used in the ngrx examples (https://ngrx.io/guide/effects, collections.effect)

File: projects/media-viewer/src/lib/store/store-utils.ts (hunk 5 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/store/store-utils.ts:92]: no, intelliJ flagged it as not used

File: projects/media-viewer/src/lib/toolbar/main-toolbar/sub-toolbar/sub-toolbar.component.html (hunk 7 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/toolbar/main-toolbar/sub-toolbar/sub-toolbar.component.html:2]: ok

File: projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/side-bar.component.ts (hunk 8 above)
Comment (benouaer) [reply projects/media-viewer/src/lib/viewers/pdf-viewer/side-bar/side-bar.component.ts:56]: it's the text of the new bookmark. it's like a placeholder when they click the bookmark button, and then they can change that it

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 9]
```diff
@@ -83,15 +88,19 @@ export class AnnotationSetComponent implements OnInit, OnDestroy {
...
-    const selection = window.getSelection().toString();
-    this.viewerEvents.createBookmarkEvent.next({
-      name: selection.length > 0 ? selection : 'new bookmark',
-      pageNumber: `${this.highlightPage - 1}`,
-      xCoordinate: rectangle.x,
-      yCoordinate: rectangle.y
-    });
-    this.highlightService.resetHighlight();
-    this.rectangles = undefined;
+    this.store.pipe(select(fromBookmarks.getBookmarkInfo), take(1))
```
Comment (fazanki) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts:91]: unsubscribe

**Pull request merged**